packages/noya-state/src/selectors/elementSelectors.ts

```diff
@@ -21,12 +21,30 @@ type PathElement =
       value: SelectedControlPoint;
     };
 
+export function canClosePath(state: ApplicationState, element: PathElement) {
+  if (element.type === 'point') {
+    const [layerId, index] = element.value;
+
+    const layers = getSelectedLayers(state)
+      .filter(Layers.isPointsLayer)
+      .filter((layer) => layer.do_objectID === layerId);
+
+    return (
```

Comment:
This is close! We only want to close the path if the first/last point is selected, but at the moment this lets us close the path if _any_ other point is selected.

It does a little extra work, but it'd probably be best to use `getIndexPathOfOpenShapeLayer` to determine if we're in a scenario where we can close a path, since we know that works correctly and then we'll have consistent behavior everywhere.

```ts
// A path needs at least 2 points to be closed.
// TODO: Don't allow closing a straight line segment with 2 points.
if (layer.points.length < 2) return false

const selectedPoint = getIndexPathOfOpenShapeLayer(state)

if (!selectedPoint) return false

const lastIndex = layer.points.length - 1

return (
  (index === lastIndex && selectedPoint.pointIndex === 0) ||
  (index === 0 && selectedPoint.pointIndex === lastIndex)
)
```

I threw in a TODO for the special case of a 2-point path, but we don't need to handle that right now